drivers/audio/max98091.c

```diff
@@ -0,0 +1,354 @@
+{
+	uint8_t psclk, mclk;
+
+	if (mclk_freq >= 10000000 && mclk_freq <= 20000000) {
+		psclk = M98091_PSCLK_DIV1;
+	} else if (mclk_freq > 20000000 && mclk_freq <= 40000000) {
+		psclk = M98091_PSCLK_DIV2;
+	} else if (mclk_freq > 40000000 && mclk_freq <= 60000000) {
+		psclk = M98091_PSCLK_DIV4;
+	} else {
+		LOG_ERR("Invalid MCLK frequency: %u", mclk_freq);
+		return;
+	}
+	max98091_write_reg(dev, M98091_REG_SYSTEM_CLOCK, psclk);
+	LOG_DBG("System clock set: PSCLK=0x%02x", psclk);
+
+	max98091_update_reg(dev, M98091_REG_MASTER_MODE, M98091_MAS_MASK, 0);
+}
+
+static int max98091_set_volume_or_mute(const struct device *dev, audio_channel_t channel, int value,
+				      bool is_volume)
+{
+	uint8_t hp_mask = is_volume ? M98091_HPVOLL_MASK : M98091_HPLM_MASK;
+	uint8_t spk_mask = is_volume ? M98091_SPVOLL_MASK : M98091_SPLM_MASK;
+
+	switch (channel) {
+	case AUDIO_CHANNEL_FRONT_LEFT:
+		max98091_update_reg(dev, M98091_REG_LEFT_SPK_VOLUME, spk_mask, value);
+		return 0;
+
+	case AUDIO_CHANNEL_FRONT_RIGHT:
+		max98091_update_reg(dev, M98091_REG_RIGHT_SPK_VOLUME, spk_mask, value);
+		return 0;
+
+	case AUDIO_CHANNEL_HEADPHONE_LEFT:
+		max98091_update_reg(dev, M98091_REG_LEFT_HP_VOLUME, hp_mask, value);
+		return 0;
+
+	case AUDIO_CHANNEL_HEADPHONE_RIGHT:
+		max98091_update_reg(dev, M98091_REG_RIGHT_HP_VOLUME, hp_mask, value);
+		return 0;
+
+	case AUDIO_CHANNEL_ALL:
+		max98091_update_reg(dev, M98091_REG_LEFT_SPK_VOLUME, spk_mask, value);
+		max98091_update_reg(dev, M98091_REG_RIGHT_SPK_VOLUME, spk_mask, value);
+		max98091_update_reg(dev, M98091_REG_LEFT_HP_VOLUME, hp_mask, value);
+		max98091_update_reg(dev, M98091_REG_RIGHT_HP_VOLUME, hp_mask, value);
+		return 0;
+
+	default:
+		return -EINVAL;
+	}
+}
+
+static int max98091_out_volume_config(const struct device *dev, audio_channel_t channel, int volume)
+{
+	return max98091_set_volume_or_mute(dev, channel, volume, true);
+}
+
+static int max98091_out_mute_config(const struct device *dev, audio_channel_t channel, bool mute)
+{
+	return max98091_set_volume_or_mute(dev, channel, mute, false);
+}
+
+/* Audio Path Configuration */
+static void max98091_configure_output(const struct device *dev)
+{
+	max98091_update_reg(dev, M98091_REG_IO_CONFIGURATION, M98091_SDIEN_MASK, M98091_SDIEN_MASK);
+
+	max98091_write_reg(dev, M98091_REG_LEFT_SPK_MIXER, M98091_MIXSPL_DACL_MASK);
+	max98091_write_reg(dev, M98091_REG_RIGHT_SPK_MIXER, M98091_MIXSPR_DACR_MASK);
+
+	/* select DAC only source */
+	max98091_write_reg(dev, M98091_REG_HP_CONTROL, 0x00);
+
+	/* M98091_HPREN_MASK, M98091_HPLEN_MASK, M98091_SPREN_MASK,
+	 * M98091_SPLEN_MASK, M98091_DAREN_MASK, M98091_DALEN_MASK
+	 */
+	max98091_write_reg(dev, M98091_REG_OUTPUT_ENABLE, 0xf3);
+
+	max98091_out_volume_config(dev, AUDIO_CHANNEL_ALL, M98091_DEFAULT_VOLUME);
+	max98091_out_mute_config(dev, AUDIO_CHANNEL_ALL, false);
+}
+
+static void max98091_start_output(const struct device *dev)
+{
+}
```

Comment:
Same for `max98091_stop_output`
```suggestion
static void max98091_start_output(const struct device *dev)
{
	ARG_UNUSED(dev);
}
```